rpcs3/Crypto/unedat.cpp

```diff
@@ -715,26 +716,31 @@ bool extract_all_data(const fs::file* input, const fs::file* output, const char*
 
 		// Select EDAT key.
 		if ((NPD.license & 0x3) == 0x3)           // Type 3: Use supplied devklic.
-			memcpy(&key, devklic, 0x10);
-		else if ((NPD.license & 0x2) == 0x2)      // Type 2: Use key from RAP file (RIF key).
 		{
-			memcpy(&key, rifkey, 0x10);
+			std::memcpy(&key, devklic, 0x10);
+		}
+		else // Type 2: Use key from RAP file (RIF key). (also used for type 1 at the moment)
+		{
+			const std::string rap_path = rpcs3::utils::get_rap_file_path(NPD.content_id);
 
-			// Make sure we don't have an empty RIF key.
-			if (!key)
+			if (fs::file rap{rap_path}; rap && rap.size() >= sizeof(key))
 			{
-				edat_log.error("A valid RAP file is needed for this EDAT file! (local activation)");
-				return true;
+				key = GetEdatRifKeyFromRapFile(rap);
+			}
+			else if (rifkey)
+			{
+				// Fallback
+				if (rifkey)
```

Comment:
double check

Reply:
I don't know why it even exists as a param if it's not used anymore

Reply:
Removed.